src/v2/Components/NavBar/NavItem.tsx

```diff
@@ -67,81 +128,92 @@ export const NavItem: React.FC<NavItemProps> = ({
     }
   }
 
+  const containerRef = useRef<null | HTMLDivElement>(null)
+
+  // Close the subnav if it is open and our focus moves outside of it
+  useEffect(() => {
+    const checkFocus = () => {
+      if (!containerRef?.current) return // Not mounted
+      if (!isVisible) return // Isn't open
+      if (containerRef.current.contains(document.activeElement)) return // Focus is within sub-nav
+
+      setIsVisible(false)
+    }
+
+    document.addEventListener("focus", checkFocus, true)
+    return () => {
+      document.removeEventListener("focus", checkFocus)
+    }
+  }, [isVisible])
+
+  const handleClick = () => {
+    onClick && onClick()
+    !!Menu && setIsVisible(prevIsVisible => !prevIsVisible)
+    trackClick()
+  }
+
+  // Add a very short delay to avoid trapping the mouse and capturing intent
+  const openDelayRef = useRef<null | ReturnType<typeof setTimeout>>(null)
+  // Add a slightly longer delay to improve UX when moving at an angle off HitArea
+  const closeDelayRef = useRef<null | ReturnType<typeof setTimeout>>(null)
```

Comment:
Ah, this is a nice non-DOM use for refs 👍 

This brings back [memories](https://github.com/artsy/reaction/pull/3369#pullrequestreview-391252596). (My own attempt to fix this was _not_ successful 🙈 ). I really like your solution.